### PR TITLE
Software

### PR DESCRIPTION
**Pull request by NMoksnes**
Hi Will!
Here is the blog post for software releases ready for you to review.
/Nandi

### CODE REVIEW

File: _posts/2022-03-13-software.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  
+![image](https://user-images.githubusercontent.com/30128518/159683921-d979dd9c-0db3-4f99-b581-c62870562cab.png)
+
+A software release covers all of the principles for open science and a brief decsription can be found below and longer decription can be found in the respective principle.
+  
+### [Workflow](https://climatecompatiblegrowth.github.io/guidelines/open%20science/2022/03/18/workflow.html)
+When developing a workflow for a software release it is important to develop clear objects and steps with tests so that bugs and issues can be identified and reported. This is to assure that the workflow is working as intended at the release and that bugs or issues can be fixed for next release. More advanced software development has tests inplace which are checked for each change in code to ensure consistency. One such integrated tool on Github is [Travis CI](https://docs.travis-ci.com/user/tutorial/) which builds the code and tests it with user defined tests. Travis CI has a payment plan, but for open source accounts it is free under travis-ci.com.
```
Comment (willu47) [inline _posts/2022-03-13-software.md:19]: I think here it is more important to zoom out and provide the context to the reader. Why would you want to release a workflow? Who would it help? How does it support Open Science and reproducible research?

File: _posts/2022-03-13-software.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  
+### [Version control](https://climatecompatiblegrowth.github.io/guidelines/)
+The software is first built using version control (such as [Git](https://git-scm.com/), Mercurial or Subversion). The version control tracks all the changes that you make and allows to also revert back to older versions. The version control system allows for collaborative work as each contributor/author can create an own "branch" and develop new code. The new code can be reviewed and compared to the other developments and "merged" to the main branch if the new developement is accepted. When the code is ready and tested it can be released.
+
+### [License](https://climatecompatiblegrowth.github.io/guidelines/practice/2022/03/01/licensing.html)
```
Comment (willu47) [inline _posts/2022-03-13-software.md:24]: I suggest shortening this (e.g. remove the details of three license types)

File: _posts/2022-03-13-software.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+- Free-for-all licenses (permissive) This type of licenses require to acknowledge the original author(s) of the licensed work in any derivative work. However, it allows derivative works to be then published under a different type of license, and/or made proprietary.
+
+- Keep-open licenses This type of licenses require for the derivative work to be also released under open license. However, the derivative work (e.g. a software product) could then be included as one part of a larger proprietary work.
+
+- Share-alike (copyleft) licenses This type of licenses require that, if the licensed work is included in larger work or project, the entire work should be released open-source.
+
+For more information on open source licenses you can find more details [on Open Source initiative](https://opensource.org/licenses).
+
+### [Metadata](https://climatecompatiblegrowth.github.io/guidelines/)
+Metadata is "data about the data" meaning that it describes the context, quality and condition of a e.g. dataset.
```
Comment (willu47) [inline _posts/2022-03-13-software.md:36]: Main points here are that metadata:

- are machine readable, so the software release can be indexed (e.g. on PyPI, or Zenodo) [findable]
- follow a standard depending on where the software is released

File: _posts/2022-03-13-software.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+### [Release](https://climatecompatiblegrowth.github.io/guidelines/)
+A software release can be managed through for example [Github relases](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository). On Github you first make a tag to mark that this version of the code is a specific version. If you also want to make a release of the code you can then create a release based on that tag.
+
+If you make a release on GitHub then this can be linked to your Zenodo account which will give the software a DOI and also make it citable for each release you make.
+  
+## Example of software release - NISMOD
+  
+In the CCG project a Python package has been released for NISMOD [snail](https://github.com/nismod/snail/releases) under different versions using [Python Package Index (PyPi)](https://pypi.org/).
```
Comment (willu47) [inline _posts/2022-03-13-software.md:49]: We might want to think how to improve this example.  Perhaps the PyPSA example would be more descriptive as it works across several of the practices and outputs.

**Review by willu47 — CHANGES_REQUESTED** [review]
Nice @NMoksnes!  I've made a few suggestions.

File: _posts/2022-03-13-software.md (hunk 4 above)
Comment (NMoksnes) [reply _posts/2022-03-13-software.md:48]: I have the PyPSA example in Workflows, but I can add it here as well if you want.

File: _posts/2022-03-13-software.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title:  "Data"
```
[the remark this answers is not in the document]
Comment (NMoksnes) [reply _posts/2022-03-13-software.md:3]: OK

File: _posts/2022-03-13-software.md (hunk 1 above)
Comment (NMoksnes) [reply _posts/2022-03-13-software.md:17]: I added a sentence on Open science and resuability. Let me know if that is more what you had in mind.

File: _posts/2022-03-13-software.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  
+### [Version control](https://climatecompatiblegrowth.github.io/guidelines/)
+The software is first built using version control (such as [Git](https://git-scm.com/), Mercurial or Subversion). The version control tracks all the changes that you make and allows to also revert back to older versions. The version control system allows for collaborative work as each contributor/author can create an own "branch" and develop new code. The new code can be reviewed and compared to the other developments and "merged" to the main branch if the new developement is accepted. When the code is ready and tested it can be released.
```
[the remark this answers is not in the document]
Comment (NMoksnes) [reply _posts/2022-03-13-software.md:20]: I added two sentences on this. Hope it clarifies.

File: _posts/2022-03-13-software.md (hunk 3 above)
Comment (NMoksnes) [reply _posts/2022-03-13-software.md:34]: OK

**Review by willu47 — APPROVED** [review]

**Pull request merged**